internal/connectors/plugins/public/adyen/client/accounts.go

```diff
@@ -0,0 +1,29 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/adyen/adyen-go-api-library/v7/src/management"
+)
+
+func (c *client) GetMerchantAccounts(ctx context.Context, pageNumber, pageSize int32) ([]management.Merchant, error) {
+	// TODO(polo): add metrics
+	// f := connectors.ClientMetrics(ctx, "adyen", "list_merchant_accounts")
+	// now := time.Now()
+	// defer f(ctx, now)
+
+	listMerchantsResponse, raw, err := c.client.Management().AccountMerchantLevelApi.ListMerchantAccounts(
+		ctx,
+		c.client.Management().AccountMerchantLevelApi.ListMerchantAccountsInput().PageNumber(pageNumber).PageSize(pageSize),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if raw.StatusCode >= 300 {
```

Comment:
3xx status are generally successful redirects. Would be better to use `http.StatusBadRequest` as the border condition

Reply:
Alright, i've made the change thanks !